main.cpp

```diff
@@ -45,28 +45,480 @@ send me a DM to check your pull request
  Wait for my code review.
  */
 
+#include <iostream>
+
 /*
  copied UDT 1:
  */
 
+struct HarmonicSet
+{
+    HarmonicSet(float bass, float gen);
+    HarmonicSet();
+    ~HarmonicSet();
+
+    float bassFrequency, genFrequency;
+    float current, previous;
+
+    void playSet(float minFrequency, float maxFrequency);
+
+    float calculateHarmonicity();
+    float next();
+    float prev();
+    void reset();
+
+    void playFiltered(float minFrequency, float maxFrequency, int genRule);
+};
+
+HarmonicSet::HarmonicSet(float bass, float gen)
+{
+    bassFrequency = bass;
+    genFrequency = gen;
+    current = gen;
+    previous = bass;
+}
+
+HarmonicSet::HarmonicSet()
+{
+    bassFrequency = 110.f;
+    genFrequency = 220.f;
+    reset();
+}
```

Comment:
You can move that default constructor and instead give your constructor arguments these default values.
```
HarmonicSet::HarmonicSet(float bass = 110.f, float gen = 220.f) : 
bassFrequency(bass), 
genFrequency(gen), 
current(gen), 
previous(bass)
{
    reset();
}
```
always favor the member initializer list over initializing member variables in the constructor body.
- [ ] revise